test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -31,20 +31,22 @@
 
 (deftest ^:integration legacy-routes
   (testing "The legacy web routing service properly handles old routes."
-    (bootstrap/with-puppetserver-running app
-      {}
-      (is (= 200 (:status (http-get "/v2.0/environments"))))
-      (is (= 200 (:status (http-get "/production/node/localhost"))))
-      (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
+    (bootstrap/with-puppetserver-running-with-mock-jrubies
+     app
+     {}
+     (is (= 200 (:status (http-get "/v2.0/environments"))))
+     (is (= 200 (:status (http-get "/production/node/localhost"))))
+     (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
 
 (deftest ^:integration old-master-route-config
   (testing "The old map-style route configuration map still works."
-    (bootstrap/with-puppetserver-running app
-      {:web-router-service
-       {:puppetlabs.services.master.master-service/master-service
-        {:master-routes "/puppet"
-         :invalid-in-puppet-4 "/"}}}
-      (is (= 200 (:status (http-get "/puppet/v3/node/localhost?environment=production"))))))
+    (bootstrap/with-puppetserver-running-with-mock-jrubies
+     app
+     {:web-router-service
+      {:puppetlabs.services.master.master-service/master-service
+       {:master-routes "/puppet"
+        :invalid-in-puppet-4 "/"}}}
+     (is (= 200 (:status (http-get "/puppet/v3/node/localhost?environment=production"))))))
```

Comment:
I guess maybe `static_file_content` could be used here instead but then we'd probably need to add a lot of other configuration to support it as well.

Sounds like what I need to do is change `handleRequest` in the mock JRubyPuppet to throw an exception and, for every test that fails as a result, change the test back to using a real JRuby instead.  Kind of a bummer, but I'll do it.